modules/standard/Errors.chpl

```diff
@@ -126,6 +126,15 @@ module Errors {
     }
   }
 
+  class KeyNotFoundError : Error {
```

Comment:
1. This should be documented. It is unfortunate that some existing ones aren't documented either.
2. More importantly, there are questions around this type that should be answered under the library stabilization roof:
  a. Should this be in standard/Errors given that it is pretty map-specific?
  b. Are there other errors thrown from other data structures? If so, what do they look like?

For now, I would document and move this to the `Map` module as it feels more future-proof to me. I think we'll do another round of review on the `Errors` module. I think the design questions here should be discussed in that review.

Reply:
> Should this be in standard/Errors given that it is pretty map-specific?

While at the moment this has no other uses, it could also apply to other data structures (heap, list, set, etc.) that are doing the same thing if we choose to standardize this `throws` approach. There is a design discussion [here](https://github.com/chapel-lang/chapel/issues/18825) about this issue.

> Are there other errors thrown from other data structures? If so, what do they look like?

No, there are not, all of the others currently halt today. In the design discussion around this issue [here](https://github.com/chapel-lang/chapel/issues/18786), I mentioned that this felt inconsistent to me with the other modules, but the others who participated (pretty much just Brad) seemed to think that moving to throw would probably be better and that they didn't have a problem with doing it only in this single location for now.

> I would document and move this to the Map module

Will do, that makes sense to me.

Reply:
All makes sense. Thanks for the context!

Reply:
Maybe you can put a link to that issue in the PR/merge message?